commit 96082a499a8e0b861e13cead287dd01858315634 (refs/changes/54/4454/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-06-28T23:35:17+00:00 (1 year, 3 months ago)
    
    joyent/node-cueball#142 want ability to set decoherence/shuffle timer interval
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/docs/api.adoc b/docs/api.adoc
index ae99c37..54392f6 100644
--- a/docs/api.adoc
+++ b/docs/api.adoc
@@ -248,6 +248,10 @@ Parameters
     connections
   * `targetClaimDelay` -- optional Number, a target time for latency to claim
     accept/timeout, done through Adaptive Queue Management
+  * `decoherenceInterval` -- optional Number, seconds between re-shuffles of
+    the backend preference list to try to avoid "coherence" between multiple
+    clients (see the Cueball Internals document for more information). Minimum
+    value of 60 (smaller values will be clamped to 60 seconds).
   * `resolver` -- optional instance of an object meeting the Resolver interface
     below.  You would typically obtain this object by either creating your own
     Resolver directly or using the `resolverForIpOrDomain` function.
diff --git a/lib/pool.js b/lib/pool.js
index 4cf6678..86e5f77 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -221,10 +221,17 @@ function CueBallConnectionPool(options) {
 	}, 10000);
 	this.p_rebalTimerInst.unref();
 
+	mod_assert.optionalFinite(options.decoherenceInterval,
+	    'options.decoherenceInterval');
+	var shuffleIntvl = options.decoherenceInterval;
+	if (shuffleIntvl === undefined || shuffleIntvl === null ||
+	    shuffleIntvl < 60) {
+		shuffleIntvl = 60;
+	}
 	this.p_shuffleTimer = new EventEmitter();
 	this.p_shuffleTimerInst = setInterval(function () {
 		self.p_shuffleTimer.emit('timeout');
-	}, 60000);
+	}, shuffleIntvl * 1000);
 	this.p_shuffleTimerInst.unref();
 
 	this.p_lastRebalClamped = false;
diff --git a/package.json b/package.json
index cc6cbf3..9f7c420 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "cueball",
-  "version": "2.5.3",
+  "version": "2.6.0",
   "description": "manage a pool of connections to a multi-node service where nodes are listed in DNS",
   "main": "lib/index.js",
   "dependencies": {
